go-controller/pkg/metrics/ovnkube_controller.go

```diff
@@ -736,7 +736,7 @@ func (pr *PodRecorder) Run(sbClient libovsdbclient.Client, stop <-chan struct{})
 		prometheus.MustRegister(metricPortBindingChassisLatency)
 	})
 
-	pr.queue = workqueue.New()
+	pr.queue = workqueue.NewTyped[*item]()
```

Comment:
Out of curiosity, why is it a pointer?

Reply:
It was already a pointer
```
func (pr *PodRecorder) processNextItem() bool {
	i, term := pr.queue.Get()
	if term {
		return false
	}
	pr.processItem(i)
	pr.queue.Done(i)
	return true
}

func (pr *PodRecorder) processItem(i *item) {
```

If you are asking why, I can only guess to save on copying, since the memory is ours anyway as it is us pushing into the queue as well. But cannot really say much about it.